cognite/client/data_classes/simulators/models.py

```diff
@@ -323,3 +354,57 @@ def _get_update_properties(cls, item: CogniteResource | None = None) -> list[Pro
             PropertySpec("name"),
             PropertySpec("description"),
         ]
+
+
+@dataclass
+class SimulatorModelExternalDependencyFile(CogniteObject):
+    """
+    Represents the CDF file associated with the external dependency.
+    Args:
+        id (int): The id of the file associated with the external dependency.
+    """
+
+    id: int
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            id=resource["id"],
+        )
+
+
+@dataclass
+class SimulatorModelRevisionExternalDependency(CogniteObject):
+    """
+    Represents an external dependency for a simulator model revision.
+    Args:
+        file (SimulatorModelExternalDependencyFile): The CDF file associated with the external dependency.
+        arguments (dict[str, str]): A dictionary that contains the key-value pairs (fields) for the external dependency.
+    """
+
+    file: SimulatorModelExternalDependencyFile
+    arguments: dict[str, str]
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            file=SimulatorModelExternalDependencyFile._load(resource["file"], cognite_client),
+            arguments=resource["arguments"],
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case=camel_case)
+        output["file"] = self.file.dump(camel_case=camel_case)
+        return output
+
+    @classmethod
+    def _load_list(
+        cls, resource: dict[str, Any] | list[dict[str, Any]], cognite_client: CogniteClient | None = None
```

Comment:
same question on why this can't just be
```suggestion
        cls, resource: list[dict[str, Any]], cognite_client: CogniteClient | None = None
```